Add shortNames to the AddonsLayer CRD

sbernheim opened the pull request:
Adding `al`, `layer` and `addonlayer` shortName aliases to the AddonsLayer Custom Resource Definition.

paulcarlton-ww reviewed (changes requested):
my mistake, this file is generated by kubebuilder from https://github.com/fidelity/kraan/blob/master/pkg/api/v1alpha1/addons_types.go so we should not change this. Let me fold this into the printer defaults work I'm doing

sbernheim commented:
Corrected - I've added the shortName marker to the `kubebuilder:resource` annotation on the AddonsLayer type.  You can merge into your branch or just steal it and nuke this branch if you like.
